eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1456,7 +1461,7 @@ protected function internalPublishVersion(APIVersionInfo $versionInfo, $publicat
 
         $metadataUpdateStruct = new SPIMetadataUpdateStruct();
         $metadataUpdateStruct->publicationDate = isset($publicationDate) ? $publicationDate : time();
```

Comment:
I would rather remove the changes in `publishVersion` and do something like this on this line to make sure all use of this have the same behaviour everywhere, and be closer to how legacy behaves:
```php
if ($publicationDate === null && $versionInfo->getContentInfo()->publishedDate->getTimestamp() === 0) {
    $publicationDate = time();
}

$metadataUpdateStruct = new SPIMetadataUpdateStruct();
$metadataUpdateStruct->publicationDate = $publicationDate;
```

php doc something like:
```php
@param int|null $publicationDate If null existing date is kept if there is one, otherwise current time is used.
```